Sources/PackageDescription/BuildSettings.swift

```diff
@@ -414,15 +414,15 @@ public struct SwiftSetting: Sendable {
     /// - Since: First available in PackageDescription 6.0.
     ///
     /// - Parameters:
-    ///   - version: The Swift language version to use.
+    ///   - mode: The Swift language mode to use.
     ///   - condition: A condition that restricts the application of the build setting.
     @available(_PackageDescription, introduced: 6.0)
-    public static func swiftLanguageVersion(
-      _ version: SwiftVersion,
+    public static func swiftLanguageMode(
```

Comment:
This needs to introduce a new function and mark the old one as deprecated, otherwise existing `Package.swift` manifests will break.

Reply:
I'm willing to do so, but this would be an API change within the same release cycle. No shipping release version of Swift PM has the original API.

Reply:
I think we should do this anyway - it would be nice not to break packages that have already started using this in their conversion to Swift 6. I admit it's a little weird, but IMO worth handling.

Reply:
Assuming this proposal does get reviewed in the Swift 6.0 timeframe, I could completely see adding an 'obsoleted in 6 / renamed in 6' for `swiftLanguageVersion` to give folks an easy migration to move over during pre-release Swift 6.0 with a fix-it - but removing that for the final version so it doesn't have to be baked into the API for all time.

Reply:
> but removing that for the final version so it doesn't have to be baked into the API for all time

We can just deprecate it and then remove when we remove the other `swiftLanguageVersion` 🤷